lib/workers/repository/update/branch/auto-replace.ts

```diff
@@ -26,7 +26,27 @@ export async function confirmIfDepUpdated(
     // istanbul ignore if
     if (!newExtract) {
       // TODO: fix types (#22198)
-      logger.debug(`Could not extract ${packageFile!}`);
+      logger.debug(
+        `Could not extract ${packageFile!} (manager=${manager}) after autoreplace. Did the autoreplace make the file unparseable?`
+      );
+      logger.trace(
+        { packageFile, content: newContent },
+        'packageFile content after autoreplace'
+      );
+      return false;
+    }
+    // istanbul ignore if
+    if (!newExtract.deps?.length) {
+      logger.debug(
+        `Extracted ${packageFile!} after autoreplace has no deps array. Did the autoreplace make the file unparseable?`
+      );
+      return false;
+    }
+    // istanbul ignore if
+    if (is.number(depIndex) && depIndex >= newExtract.deps.length) {
```

Comment:
this is  only truthy if we're on some latest index. we don't catch if dep index is 0 but the length of newly extracted deps has changed. 🤔

Reply:
Yes, but it's at least more verbose than the current approach where it resolves to `undefined`